src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -38,31 +38,32 @@ namespace nupic {
 
 /**
  * This makes a deep copy of the buffer so this class will own the buffer.
- *
- * Note: for NTA_BasicType_SDR, the count variable is ignored. Its size is
- *       taken from the internal SDR object.
  */
-template <typename T>
-ArrayBase::ArrayBase(NTA_BasicType type, T *buffer, size_t count) {
+ArrayBase::ArrayBase(NTA_BasicType type, void *buffer, size_t count) {
   if (!BasicType::isValid(type)) {
     NTA_THROW << "Invalid NTA_BasicType " << type
               << " used in array constructor";
   }
   type_ = type;
-  if (typeid(T) == typeid(SDR)) {
-    auto dim = buffer->dimensions;
-    allocateBuffer(dim);
-    if (count > 0) {
-      std::memcpy((char *)getBuffer(), (char *)buffer,
-              count * BasicType::getSize(type));
-    }
-  } else {
-    allocateBuffer(count);
-    if (count > 0) {
-      std::memcpy((char *)getBuffer(), (char *)buffer,
-              count * BasicType::getSize(type));
-    }
+  allocateBuffer(count);
+  if (has_buffer()) {
+    std::memcpy((char *)getBuffer(), (char *)buffer,
```

Comment:
> can this be part of allocateBuffer() already?

don't understand your question.  allocateBuffer() just creates the space.  There are lots of ways in which it might be populated.

Reply:
I saw it later, I had thought the memcpy part was alwasy the same.